Source/WebKit/UIProcess/WebProcessCache.cpp

```diff
@@ -131,9 +135,15 @@ bool WebProcessCache::addProcess(std::unique_ptr<CachedProcess>&& cachedProcess)
         WEBPROCESSCACHE_RELEASE_LOG("addProcess: Evicting process from WebProcess cache because a new process was added for the same domain", previousProcess->process().processID());
 
     while (m_processesPerRegistrableDomain.size() >= capacity()) {
-        auto it = m_processesPerRegistrableDomain.random();
-        WEBPROCESSCACHE_RELEASE_LOG("addProcess: Evicting process from WebProcess cache because capacity was reached", it->value->process().processID());
-        m_processesPerRegistrableDomain.remove(it);
+        // Remove the process using the most memory
```

Comment:
I don't think I agree with this change. Random eviction is our preferred cache eviction policy. Why evict a cached process simply because it is bigger? Cached processes go away on memory pressure anyway. Also, a large process means a process which populated caches with means the page load will likely be faster if we use it.

Reply:
That is a good point! Although, I also think evicting by memory size has some good features. For example, it will minimize the total memory footprint for a given cache size, and thereby minimizing the chance for jetsams. Another benefit is that the behavior will be more consistent, which will reduce the variance in benchmark runs.

Reply:
> For example, it will minimize the total memory footprint for a given cache size, and thereby minimizing the chance for jetsams. 
Cached processes get cleared on memory pressure already.

Reply:
Also, this patch adds a lot of complexity just to determine the process size and requires processes to be suspended to enter the cache. This is not a good trade-off.